finished drone list

Matyasai opened the pull request:
…heir status

Andrei-Pav committed:
added react-router and tailwindcss

Andrei-Pav committed:
added react-router and tailwindcss

Andrei-Pav committed:
removed tailwindcss, added login functionality(only local, with a json) and custom css

Andrei-Pav committed:
removed tailwindcss, added login functionality(only local, with a json) and custom css

Andrei-Pav committed:
added simple google map to main page

Andrei-Pav committed:
added simple google map to main page

Matyasai committed:
added simple list for main page in which you can see the drones and their status

Matyasai committed:
added simple list for main page in which you can see the drones and their status

tomoiagahoratiu commented on src/main.tsx:
```diff
@@ -0,0 +1,55 @@
+import React from 'react';
+import './styles/droneList.css'
+
+const drones = [
```
I would create a models directory, where i would create:
```
type Drone = {
    id: number;
    status: DroneStatus;
};
```
And the DroneStatus
`type DroneStatus = "parked" | "delivering" etc...;`

tomoiagahoratiu replied:
then i would assign the Drone type to the drones constant

Matyasai replied:
Done

tomoiagahoratiu commented on src/main.tsx:
```diff
@@ -0,0 +1,55 @@
+                {drones[i].id} - {drones[i].status};
+            </li>
+        }
+        else if(drones[i].status === "delivering") {
```
I would modify the following code:
create a switch block function and based on the drones status return the correct css class name.
then I would create the list like this 
```
const dronesList = drones.map((drone, index) => (
    <li key={index} className={getClassName(drone.status)}>
        {drone.id} - {drone.status}
    </li>
```

Matyasai replied:
Done

tomoiagahoratiu commented on src/styles/droneList.css:
```diff
@@ -0,0 +1,25 @@
+ol {
```
I would also play a little bit with the css and maybe make the list of drones to look more like the design on figma

Matyasai replied:
Will start from tomorrow

Matyasai committed:
added modules directory and Drone type

Matyasai committed:
added modules directory and Drone type

Matyasai committed:
made the code easier to read using map and switch instead of ifs and elses and for loop

Matyasai committed:
made the code easier to read using map and switch instead of ifs and elses and for loop

Matyasai committed:
changed the names of mehtods so that they match their purposes

Matyasai committed:
changed the names of mehtods so that they match their purposes

Matyasai committed:
changed style of drone list and updated drone definition

Andrei-Pav committed:
code refactoring

Andrei-Pav committed:
rebased branch; added better log in styles, for better ux

Andrei-Pav committed:
Merge branch 'login' of https://github.com/cloudcargo2024/cloudcargo-web-application into login

Andrei-Pav committed:
deleted some empty spaces

Andrei-Pav committed:
added user type, and other fixes

Andrei-Pav committed:
solved problems

Andrei-Pav committed:
fixed import

tomoiagahoratiu committed:
Merge pull request #9 from cloudcargo2024/login

Fixes for login(including rebase)

Matyasai committed:
added display of coordinates when hovering over the respective drone

Matyasai committed:
added display of coordinates when hovering over the respective drone
merging drone list with main project

Colividiu committed:
Added map element

Added the map and faked the process of the drones moving by incrementing their coordinates

Matyasai committed:
finished drone list
adding drone list to main project

Matyasai committed:
finished drone list

Colividiu committed:
Added New Logo

Self explainatory

tomoiagahoratiu commented:
Can you please rebase this pr with main?

Colividiu committed:
Merge pull request #10 from cloudcargo2024/map-API-features

Added map element

Matyasai committed:
resolved conflicts

Matyasai commented:
> Can you please rebase this pr with main?

rebased with main @tomoiagahoratiu @tomoiagahoratiu

tomoiagahoratiu commented:
there are still some issues with the PR, did you add the logic that was already added by Andrei with your commits? Because when looking at your changes you have a lot of changes related to other of your colleagues work.
